pkg/handler/enqueue_annotation.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ EventHandler = &EnqueueRequestForAnnotation{}
+
+// EnqueueRequestForAnnotation enqueues Requests based on the presence of an annotation that contains the
+// namespaced name of the primary resource. The purpose of this handler is to support cross-scope ownership
+// relationships that are not supported by native owner references and then, it should only be used in very
+// specific circumstances.
+//
+// Note that an annotation-based watch handler does not have the same owner reference restrictions.
+// However, do not provide the same feature set either. Beware that owner references are used for garbage collection
+// and then, if a resource's owner is deleted, the resource will also be deleted. However, this scenario cannot be satisfied
+// with an annotation-based watch handler and to achieve the same would be required implement finalizers to allow
+// the operator manually garbage collect the objects.
+//
+// The primary use case for this, is to have a controller enqueue requests for the following scenarios
+// 1. namespaced primary object and dependent cluster scoped resource
+// 2. cluster scoped primary object.
```

Comment:
Correct me if I'm wrong, but hasn't Kubernetes allowed cluster-scoped primary objects to own namespace-scoped child objects for a few versions now? If so, case number 2 should be removed.

Reply:
IHMO: The primary use case was not really made easier the understatement. So, I removed it and added the examples.